docs/DeveloperGuide.adoc

```diff
@@ -275,55 +275,319 @@ Refer to the guide <<Testing#, here>>.
+
+|`* *` |student who prefers visual data |visualise my income/spendings in a graph/chart |it is easier to keep track of my expenditures
+
+|`* *` |student that needs to explain their spendings to their parents |show them the chart of my everyday spendings |It is convenient and more visual
+
+|`* *` |organised student |categorise my spendings |I know the proportions of my spendings
+
+|`* *` |student with a huge wardrobe |keep track of my expenditure on clothing |I can control my shopaholic tendencies
+
+|`* *` |student on diet |track how much I spend on food |I would control myself from spending too much on food
+
+|`*` |rich student |keep track of what I bought |I can show off to my friends
+
+|`*` |rich student who always lends people money |take note of who owes me money |I can track them and ask them for it back
+
+|`*` |someone with few friends |keep track of who I paid for or who paid for me first |I know who are my friends, and the frequency I go out with them
+
+|`*` |tech-savvy loan shark |I want to conveniently record who owes me money |I can remind them to pay back through email
+
+|`*` |tech-savvy loan shark |I want a convenient way to calculate interest rate |I don't have to do it manually
+
 |=======================================================================
 
 _{More to be added}_
+//end::userStories[]
 
+//tag::useCases[]
 [appendix]
 == Use Cases
 
-(For all use cases below, the *System* is the `AddressBook` and the *Actor* is the `user`, unless specified otherwise)
+=== Wallet Tab
+
+(For all use cases below, the *System* is the `Wallet` and the *Actor* is the `User`, unless specified otherwise)
+
+[discrete]
+==== Use case: Recording an expense
+
+<<MSS, *MSS*>>
+
+1.  User keys in an expense.
+2.  <<wallet, Wallet>> adds the expense.
++
+Use case ends.
+
+<<extensions, *Extensions*>>
+
+[none]
+* 1a. The expense keyed in is invalid.
++
+[none]
+** 1a1. Wallet shows an error message.
+** 1a2. User re-enters the expense.
++
+Steps 1a1-1a2 are repeated until the expense keyed in is correct.
++
+Use case resumes at step 2.
+
+[discrete]
+==== Use case: Recording an income
+
+*MSS*
+
+1. User keys in an income.
+2. Wallet adds the income.
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 1a. The income keyed in is invalid.
++
+[none]
+** 1a1. Wallet shows an error message.
+** 1a2. User re-enters the income.
++
+Steps 1a1-1a2 are repeated until the income keyed in is correct.
++
+Use case resumes at step 2.
+
+[discrete]
+==== Use case: Setting budget
+
+*MSS*
+
+1. User keys in a budget.
+2. Wallet sets the amount keyed in as the budget of the month.
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 1a. The amount keyed in is invalid.
++
+[none]
+** 1a1. Wallet shows an error message.
+** 1a2. User re-enters the budget.
++
+Steps 1a1-1a2 are repeated until the amount keyed in is correct.
++
+Use case resumes at step 2.
+
+[discrete]
+==== Use case: Deleting an income/expense
```

Comment:
perhaps this could be split into two use cases
one for deleting income and another for deleting an expense